pkg/config/config.go

```diff
@@ -15,9 +24,18 @@ type Target struct {
 	LabelSelector string
 }
 
+// Check is a metrics threshold that should be met to consider a candidate
+// healthy.
+type Check struct {
+	Type       MetricsCheck
+	Percentile float64
+	Max        float64
+}
+
 // Strategy is the steps and configuration for rollout.
 type Strategy struct {
-	Steps []int64
+	Steps  []int64
+	Checks []Check
```

Comment:
We can probably find a better name. There's health and monitoring involved but the name doesn't reflect.

Reply:
`health-checks` or `metrics-checks`? Flagger is using `metrics` in their [config definition](https://docs.flagger.app/usage/how-it-works).

Reply:
`metrics` or `health` would be good IMO